File: src/main/java/com/mycompany/pt/io/PaymentTrackerReader.java
/*
 * |-------------------------------------------------
 * | Copyright © 2017 Colin But. All rights reserved.
 * |-------------------------------------------------
 */
package com.mycompany.pt.io;

import com.mycompany.pt.Parser;
import lombok.extern.slf4j.Slf4j;

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;

@Slf4j
public final class PaymentTrackerReader {

    public static void readInputFile(String inputFile) {
        try(BufferedReader reader = new BufferedReader(new FileReader(inputFile))) {
            String line;
            while((line = reader.readLine()) != null) {
                Parser.getInstance().invoke(line);
            }
        } catch (FileNotFoundException e) {
            log.error("File not found: {}", e);
        } catch (IOException e) {
            log.error("IO Problem: {}", e);
        }
    }
}

File: kotlin/src/main/kotlin/payment/tracker/io/PaymentTrackerReader.kt
/*
 * |-------------------------------------------------
 * | Copyright © 2019 Colin But. All rights reserved.
 * |-------------------------------------------------
 */
package payment.tracker.io

import java.io.BufferedReader
import java.io.File
import java.io.FileReader

object PaymentTrackerReader {
    fun readInputFile(file: String) : List<String> {
        return BufferedReader(FileReader(File(file))).readLines()
    }
}
